### PR TITLE
Additional documentation fixes.

### PR DESCRIPTION
**Pull request by coryan**
More of the changes for #1888.

### CODE REVIEW

**Comment by coryan** [comment]
Uploaded the generated documentation to:

https://coryan.github.io/google-cloud-cpp/latest/storage/classgoogle_1_1cloud_1_1storage_1_1v0_1_1Client.html

**Comment by coryan** [comment]
I figured out how to upload the docs for this PR too:

https://coryan.github.io/google-cloud-cpp/1.1898.0/storage/classgoogle_1_1cloud_1_1storage_1_1v0_1_1Client.html

File: google/cloud/storage/client.h [hunk 1]
```diff
@@ -1425,6 +1450,13 @@ class Client {
   /**
    * Updates the value of an existing bucket ACL.
    *
+   * @note
+   * For changing BucketAccessControl the Patch and Update APIs basically offer
```
Comment (devbww) [inline google/cloud/storage/client.h]: Opt: I might put a comma after "For changing BucketAccessControl" (et seq).

Comment (coryan) [reply google/cloud/storage/client.h]: Done.

**Comment by coryan** [comment]
@frankyn I changed the "See Also" sections. PTAL.

**Pull request merged**